src/WebJobs.Extensions.DurableTask/LocalGrpcListener.cs

```diff
@@ -154,27 +159,53 @@ public override Task<Empty> Hello(Empty request, ServerCallContext context)
             {
                 try
                 {
-                    string instanceId = request.InstanceId ?? Guid.NewGuid().ToString("N");
-                    TaskHubClient taskhubClient = new TaskHubClient(this.GetDurabilityProvider(context));
-                    OrchestrationInstance instance;
+                    // Create the orchestration instance
+                    var instance = new OrchestrationInstance
+                    {
+                        InstanceId = request.InstanceId ?? Guid.NewGuid().ToString("N"),
+                        ExecutionId = Guid.NewGuid().ToString(),
+                    };
 
-                    // TODO: Ideally, we'd have a single method in the taskhubClient that can handle both scheduled and non-scheduled starts.
-                    // TODO: the type of `ScheduledStartTimestamp` is not nullable. Can we change it to `DateTime?` in the proto file?
-                    if (request.ScheduledStartTimestamp != null)
+                    // Create the ExecutionStartedEvent
+                    ExecutionStartedEvent executionStartedEvent = new ExecutionStartedEvent(-1, request.Input)
                     {
-                        instance = await taskhubClient.CreateScheduledOrchestrationInstanceAsync(
-                            name: request.Name, version: request.Version, instanceId: instanceId, input: Raw(request.Input), startAt: request.ScheduledStartTimestamp.ToDateTime());
-                    }
-                    else
+                        Name = request.Name,
+                        Version = request.Version,
+                        OrchestrationInstance = instance,
+                        ScheduledStartTime = request.ScheduledStartTimestamp?.ToDateTime(),
+                    };
+
+                    // Get the parent trace context from CreateInstanceRequest
+                    string? traceParent = request.ParentTraceContext.TraceParent;
+                    string? traceState = request.ParentTraceContext.TraceState;
+
+                    // Create a new activity with the parent context
+                    ActivityContext.TryParse(traceParent, traceState, out ActivityContext parentActivityContext);
+                    Activity? scheduleOrchestrationActivity = StartActivityForNewOrchestration(executionStartedEvent, parentActivityContext);
```

Comment:
Rely on dispose for stopping an activity. You can remove the `.Stop()` call with this change,
```suggestion
                    using Activity? scheduleOrchestrationActivity = StartActivityForNewOrchestration(executionStartedEvent, parentActivityContext);
```